### ISSUE TITLE
Training script to replicate results from paper?

### ISSUE THREAD

**Issue by miloknowles** [CVLAB-Unibo/Real-time-self-adaptive-deep-stereo#43]
I'm trying to train MADNet from scratch so that I can compare it to an implementation that I made in Pytorch. Is there a training script that replicates the parameters used in your paper?

I have a few specific questions as well:
- Do you use ```--lossType mean_l1``` or ```--lossType sum_l1```? The paper seems to indicate that it should be ```sum_l1```, but this results in a huge loss value and I'm guessing ```mean_l1``` is correct.
- Do you train with ```--lossWeights 0.005 0.01 0.02 0.08 0.32``` as stated in the paper? These cause an exception to be thrown, since there are 6 predicted disparity maps. If I add another weight to have 6, then I hit the exception:

```python
if args.lossWeights is not None and len(args.lossWeights)==len(predictions):
  raise ValueError('Wrong number of loss weights provide, should provide {}'.format(len(predictions)))
```

- What batch size do you train from scratch with?

**Comment by AlessioTonioni** [comment]
Unfortunately I don't have the orginal script any more but I can reply to your questions.

+ `lossType` should be `sum_l1`
+ `lossWeights` should be `0.32 0 0.08 0.02 0.01 0.005`
+ `batchSize` the bigger the better

**Comment by miloknowles** [comment]
Thanks! And just to make sure I understand, those ```lossWeights``` are in order of decreasing pyramid resolution? (i.e 0.32 corresponds to full resolution, no loss for 1/2 resolution, 0.005 weight for 1/64th resolution?)

**Comment by AlessioTonioni** [comment]
Sorry you are right I mixed up the order of the weights, editing my response above and expanding down here:

In my implementation the higher resolution is actually 1/2 of the input (which is later upsampled). At half resolution I have two predictions, one before the refiner network one after, to sum up the weights should be:

+ 0.32 * sum_l1@1/32 resolution
+ 0.08 * sum_l1@1/16 resolution
+ 0.02 * sum_l1@1/8 resolution
+ 0.01 * sum_l1@1/4 resolution
+ 0 * sum_l1@1/2 resolution before refiner
+ 0.005 * sum_l1@1/2 resolution after refiner

**Comment by miloknowles** [comment]
It looks like you do refinement after the ```G2``` module here and in the paper, which is at 1/4 resolution if I'm not mistaken? Are you saying that I should actually have a ```G1``` module that does warping and disparity prediction at 1/2 resolution and put the refinement module after that? So if the input KITTI image is cropped to ```[320,1216]```, I would be doing refinement at ```[160, 608]```?

**Comment by AlessioTonioni** [comment]
No sorry you are right again, the higher resolution is 1/4.

Too much time has passed since I wrote this code. 

I modified the message above with the proper scales.
